pkg/apiservers/openshiftapiserver.go

```diff
@@ -0,0 +1,225 @@
+package apiservers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorV1 "github.com/openshift/api/operator/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var openshiftAPIServerObjName = "cluster"
+
+// OpenshiftAPIServerBuilder provides struct for openshiftAPIServer object.
+type OpenshiftAPIServerBuilder struct {
+	// OpenshiftAPIServer definition. Used to create an openshiftAPIServer object.
+	Definition *operatorV1.OpenShiftAPIServer
+	// Created openshiftAPIServer object.
+	Object *operatorV1.OpenShiftAPIServer
+	// apiClient opens api connection to the cluster.
+	apiClient goclient.Client
+	// Used in functions that define or mutate openshiftAPIServer definition. errorMsg is processed before the
+	// OpenshiftApiServer object is created.
+	errorMsg string
+}
+
+// Exists checks whether the given openshiftAPIServer exists.
+func (builder *OpenshiftAPIServerBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect openshiftAPIServer object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// PullOpenshiftAPIServerBuilder pulls existing openshiftApiServer from the cluster.
+func PullOpenshiftAPIServerBuilder(apiClient *clients.Settings) (*OpenshiftAPIServerBuilder, error) {
+	builder := OpenshiftAPIServerBuilder{
```

Comment:
Can you add a check here to ensure apiClient isn't nil? (ref: https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/lca/imagebasedupgrade.go#L75)